src/server.ts

```diff
@@ -564,6 +564,20 @@ export class Server extends EventEmitter {
         return result;
     }
 
+    /**
+     * Forcibly close a specific pending proxy connection.
+     */
+    closeConnection(connectionId: unknown): void {
+        this.log(null, 'Closing pending socket');
+
+        const socket = this.connections.get(connectionId);
+        if (!socket) return;
+
+        socket.destroy();
```

Comment:
Destroy actually doesn't destroy the socket (however it tries to close it). See https://github.com/nodejs/node/issues/27428